airflow/models/dag.py

```diff
@@ -3442,6 +3455,7 @@ def _get_or_create_dagrun(
         conf=conf,
         data_interval=data_interval,
         triggered_by=triggered_by,
+        dag_version_id=dag_version.id if dag_version else None,
```

Comment:
This particular if else is necessary because of unit tests on dag, mappedoperator etc. We should be able to test those without creating a Dagversion